packages/adblocker/src/engine/bucket/preprocessor.ts

```diff
@@ -0,0 +1,149 @@
+import { StaticDataView } from '../../data-view';
+import IFilter from '../../filters/interface';
+import Preprocessor, { Env } from '../../preprocessor';
+
+export default class PreprocessorBucket {
+  public static deserialize(view: StaticDataView): PreprocessorBucket {
+    const preprocessors: Preprocessor[] = [];
+
+    for (let i = 0, l = view.getUint32(); i < l; i++) {
+      preprocessors.push(Preprocessor.deserialize(view));
+    }
+
+    return new this({
+      preprocessors,
+    });
+  }
+
+  private readonly preprocessors: Preprocessor[];
+  private readonly excluded: Set<number>;
+
+  constructor({
+    env = new Env(),
+    preprocessors = [],
+  }: {
+    env?: Env;
+    preprocessors?: Preprocessor[];
+  }) {
+    this.excluded = new Set();
+    this.preprocessors = preprocessors;
+
+    if (preprocessors.length) {
+      this.update({ added: preprocessors, env });
+    }
+  }
+
+  private build(env: Env) {
+    // Update excluded filter ids based on bindings
+    this.excluded.clear();
+
+    const allowed = new Set<number>();
```

Comment:
Nit. If we use `excluded` as a variable name above, let's call this one: `included`.